beacon-chain/powchain/log_processing.go

```diff
@@ -72,32 +72,55 @@ func (w *Web3Service) ProcessDepositLog(depositLog gethTypes.Log) {
 	if int64(index) <= w.lastReceivedMerkleIndex {
 		return
 	}
-
 	w.lastReceivedMerkleIndex = int64(index)
 
 	// We then decode the deposit input in order to create a deposit object
 	// we can store in our persistent DB.
 	depositInput, err := helpers.DecodeDepositInput(depositData)
 	if err != nil {
-		log.Errorf("Could not decode deposit input  %v", err)
+		log.Errorf("Could not decode deposit input %v", err)
 		return
 	}
+
 	deposit := &pb.Deposit{
 		DepositData:     depositData,
 		MerkleTreeIndex: index,
 	}
+
+	validData := true
+
+	// Make sure duplicates are rejected pre-chainstart.
 	if !w.chainStarted {
-		w.chainStartDeposits = append(w.chainStartDeposits, depositData)
-	} else {
-		w.beaconDB.InsertPendingDeposit(w.ctx, deposit, big.NewInt(int64(depositLog.BlockNumber)))
+		for _, dep := range w.chainStartDeposits {
```

Comment:
This is going to cause O(n^2) insertion time. Please consider a better algorithm